src/utils/gitUtils.ts

```diff
@@ -21,10 +21,25 @@ import { getSingleRootFsPath } from './workspaceUtils';
 export type GitWorkspaceState = { repo: Repository | null, dirty: boolean, remoteRepo: ReposGetResponseData | undefined; hasAdminAccess: boolean };
 export type VerifiedGitWorkspaceState = GitWorkspaceState & { repo: Repository };
 
+// use only when using the gitApi
+export async function callWithGitErrorHandling<T>(callback: () => Promise<T>): Promise<T | null> {
+    try {
+        return await callback();
+    } catch (gitError) {
+        // ignore empty commit errors which will happen if a user initializes a blank folder or have changes in a nested git repo
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        if (!/nothing to commit/.test(gitError.stdout)) {
+            throw new GitError(gitError);
+        }
+
+        return null;
```

Comment:
I don't know how likely this is, but isn't it possible for non-git errors to be thrown by these API calls? I worry that normal errors might be lost here

Reply:
Well, it should always be throwing the error regardless of whether or not it was a non-git error.  The only time it skips the error is if there was "nothing to commit" in the stdout.

Though we should only wrapping gitApi commands with it so we don't have to do [this](https://github.com/microsoft/vscode-azurestaticwebapps/pull/314/files#diff-5bdb19eb1274f186c25c09eaf175c74b188056e2d02e678d7b9ebfd547c4e256R173-R186) all the time.

Reply:
It looks like this part will rewrite the error message if it isn't git related: https://github.com/microsoft/vscode-azurestaticwebapps/blob/c30303e02f95a3481ea53389c8a0d8d4c38c6133/src/errors.ts#L52-L54

Reply:
Well, it'll still display the error, it'll just say Git: `error string here`.  The `hint` should take `String(err)` and that regexp is just removing [husky](https://www.npmjs.com/package/husky) output.

I'm not totally sure what your concern is here though.  We should only use `callWithGitErrorHandling` specifically when calling gitApi commands.  I wouldn't add any other function calls inside of it so that the handling will only apply to Git related errors.

Are you worried that the API calls have other errors that aren't git related?  I'm pretty sure all the git API calls are `git` cli commands, so I think that we should be good.

Reply:
I can't quite put my finger on it, but I think I agree with Will. Just seems weird to transform every possible error into a git error. If you just did a little validation (maybe `'gitErrorCode' in error`) before assuming it's a "gitError", I think I'd feel better. Would help to "fix" the eslint, too